### PR TITLE
Drop CIText & update Django

### LINKED ISSUE #189: Remove CIText dependency
**Is your feature request related to a problem? Please describe.**
Core currently depends on CIText for some name fields, this isn't necessary and we should remove it.

**Describe the solution you'd like**
Replace CIText with standard text fields and use case insensitive queries instead.

**Additional context**
This isn't as urgent as SQLite now support CIText, but we're still unnecessarily breaking other databases.

### LINKED ISSUE #235: Maximum Username Length
Whilst we don't technically _need_ to limit the username length, if we don't people will abuse it.

### PR DESCRIPTION
**Pull request by jchristgit**
Move from CIText to plain char fields.

The created database migration updates the existing field (without incurring
any data loss, so this can be run safely) and creates the new database index
preventing the creation of team and users whose name only differs in their
case.

Closes #189.

Fixes #235. ~~@thebeanogamer you need to take care when rolling this out because
we may need to adjust existing entries in the database before PostgreSQL will
accept the migration.~~

### CODE REVIEW

**Comment by jchristgit** [comment]
Note: We tried to squash the migrations in a way that renders the extension obsolete for new installations, but currently we run into circular dependency errors due to a team having an owner (a member) and a member having a team.

https://docs.djangoproject.com/en/4.0/topics/migrations/#squashing-migrations has some documentation on how to deal with this, but we haven't quite gotten it to work just yet.

File: src/member/tests.py [hunk 1]
```diff
@@ -148,6 +149,12 @@ def test_patch_member(self):
         )
         self.assertEqual(response.status_code, HTTP_403_FORBIDDEN)
 
+    def test_disallows_differently_cased_but_same_username(self):
+        """A differently-cased but otherwise same username should not be allowed registration."""
+
+        self.user.username = self.admin_user.username.upper()
+        self.assertRaises(IntegrityError, self.user.save)
```
Comment (david-cooke) [inline src/member/tests.py:156]: Similar to teams, this should 400 instead of raising an Exception

**Review by david-cooke — CHANGES_REQUESTED** [review]
Looks good, just needs exception handling on register

**Review by thebeanogamer — CHANGES_REQUESTED** [review]
We have one username over 32 chars on all our deployments, let's just add a migration to truncate those.

Also, is it worth having a test for trying to register a user/team longer than 32 chars?

**Comment by jchristgit** [comment]
> We have one username over 32 chars on all our deployments, let's just add a migration to truncate those.

No. I am not a MySQL database. I will not be writing a migration that incurs data loss.

File: src/member/tests.py (hunk 1 above)
Comment (jchristgit) [reply src/member/tests.py:156]: @david-cooke this is testing the database saving part though, not the endpoint. Or do you mean that the endpoint is missing?

**Comment by jchristgit** [comment]
@thebeanogamer per your request, the migration has been altered to introduce data loss.

**Review by thebeanogamer — APPROVED** [review]
I think this is fine

File: src/member/tests.py (hunk 1 above)
Comment (david-cooke) [reply src/member/tests.py:156]: Do we handle it on registration?

Comment (jchristgit) [reply src/member/tests.py:156]: No

Comment (jchristgit) [reply src/member/tests.py:156]: There isn't really any entry point for error handling there from what I gather, so I would vote that we merge it as-is right now and wait for Sentry to throw up 🛷

**Review by david-cooke — APPROVED** [review]

**Pull request merged**